src/server/mod.rs

```diff
@@ -2,797 +2,164 @@
 //!
 //! A `Server` is created to listen on a port, parse HTTP requests, and hand
 //! them off to a `Service`.
+//!
+//! There are two levels of APIs provide for constructing HTTP servers:
+//!
+//! - The higher-level [`Server`](Server).
+//! - The lower-level [conn](conn) module.
 
 pub mod conn;
 mod service;
+mod tcp;
 
 use std::fmt;
-use std::io;
-use std::net::{SocketAddr, TcpListener as StdTcpListener};
-use std::sync::{Arc, Mutex, Weak};
+use std::net::SocketAddr;
 use std::time::Duration;
 
-use futures::task::{self, Task};
-use futures::future::{self, Either, Executor};
-use futures::{Future, Stream, Poll, Async};
-use futures_timer::Delay;
+use futures::{Future, Stream, Poll};
 use http::{Request, Response};
 use tokio_io::{AsyncRead, AsyncWrite};
-use tokio::spawn;
-use tokio::reactor::Handle;
-use tokio::net::TcpListener;
 pub use tokio_service::{NewService, Service};
 
 use body::{Body, Payload};
-use common::Exec;
-use proto;
-use self::addr_stream::AddrStream;
+use self::conn::{Http as Http_, SpawnAll};
 use self::hyper_service::HyperService;
+use self::tcp::{AddrIncoming};
 
-pub use self::conn::Connection;
 pub use self::service::{const_service, service_fn};
 
-/// A configuration of the HTTP protocol.
-///
-/// This structure is used to create instances of `Server` or to spawn off tasks
-/// which handle a connection to an HTTP server. Each instance of `Http` can be
-/// configured with various protocol-level options such as keepalive.
-#[derive(Clone, Debug)]
-pub struct Http {
-    exec: Exec,
-    http2: bool,
-    keep_alive: bool,
-    max_buf_size: Option<usize>,
-    pipeline: bool,
-    sleep_on_errors: bool,
-}
+#[deprecated(note="import from hyper::server::conn, or consider hyper::Server")]
+#[doc(hidden)]
+pub type Http = Http_;
```

Comment:
I'm undecided on whether to keep this, or remove it. I did this since it allowed providing a hint in the message that things have changed, instead of simply `hyper::server::Http is private`. But it may be annoying to keep around for all of 0.12.x...